File: CSharp/2019/A02.cs
/*
https://www.rosettacode.org/wiki/Knuth_shuffle#C.23
 */
using System.Collections.Generic;

namespace CSharp_2019
{
    /// <summary>
    /// 亂數洗牌
    /// </summary>
    public class A02
    {
        public static string[] GenArray(){
            List<string> arr = new List<string>();
            foreach(char el in "ABCD".ToCharArray()){
                for(int i = 1 ;i<14 ;i++){
                    arr.Add($"{el+i.ToString()}");
                }
            }
            return arr.ToArray();
        }
        public static T[] Shuffle<T>(T[] list){
            return B01.KnuthShuffle(list);
        }

        
    }
}

File: Web/2019/A02.js
let fn = {
    shuffle(array) {
        for (let i = array.length - 1; i > 0; i--) {
            let j = Math.floor(Math.random() * (i + 1));
            [array[i], array[j]] = [array[j], array[i]];
        }
        return array;
    },
    GenArray(){
        let arr = []
        for (x of "ABCD"){
            for (i=1 ; i < 14 ;i++){
                arr.push(x+i);
            }
        }
        return arr ;
    }
}
let A = fn.shuffle(fn.GenArray());
console.log(A,A.length);